File: app/src/main/java/com/xlzhen/sharefileserver/service/ServerRunService.java
package com.xlzhen.sharefileserver.service;

import android.app.Notification;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.app.Service;
import android.content.Context;
import android.content.Intent;
import android.graphics.drawable.Icon;
import android.os.Binder;
import android.os.Build;
import android.os.IBinder;
import android.text.Html;
import com.evgenii.jsevaluator.JsEvaluator;
import com.xlzhen.sharefileserver.R;
import com.xlzhen.sharefileserver.entity.ServerPackage;
import com.xlzhen.sharefileserver.server.MiniJsServer;
import com.xlzhen.sharefileserver.server.ServerJS;
import com.xlzhen.sharefileserver.utils.NetWorkUtils;

import java.io.DataInputStream;
import java.io.IOException;

import org.json.JSONException;
import org.json.JSONObject;

/* loaded from: classes8.dex */
public class ServerRunService extends BaseService {
    public static ServerRunService f284c;
    public static String host;
    public int rootPort = 8080;
    public int userPort = 8090;



    public static void start(Context context) {
        if (f284c == null) {
            f284c = new ServerRunService();
            context.startService(new Intent(context, ServerRunService.class));
        }
    }

    public static String getHost() {
        return host;
    }

    public final void startManagementServer() {
        host = "http://" + NetWorkUtils.getDeviceIp() + ":" + this.rootPort;
        try {
            try {
                DataInputStream open = new DataInputStream(getAssets().open("management/package.json"));
                byte[] bArr = new byte[open.available()];
                open.readFully(bArr);
                JSONObject jSONObject = new JSONObject(new String(bArr));
                ServerPackage serverPackage = new ServerPackage();
                serverPackage.setLogo(jSONObject.getString("logo"));
                serverPackage.setName(jSONObject.getString("name"));
                serverPackage.setVersion(jSONObject.getString("version"));
                serverPackage.setAuthor(jSONObject.getString("author"));
                serverPackage.setDescription(jSONObject.getString("description"));
                serverPackage.setMain(jSONObject.getString("main"));
                serverPackage.setServerPath("management");
                open.close();
                ServerJS.getInstance().init(serverPackage, new JsEvaluator(this));
                serverPackage.setMiniJsServer(new MiniJsServer(this.rootPort, serverPackage));
                try {
                    serverPackage.getMiniJsServer().start();
                } catch (IOException e2) {
                    e2.printStackTrace();
                }
                serverPackage.setPort(host);
                serverPackage.setPower(true);
            } catch (JSONException e4) {
                e4.printStackTrace();
            }
        } catch (IOException e3) {
            e3.printStackTrace();
        }
    }

    public final void startUserServer() {
        String str = "http://" + NetWorkUtils.getDeviceIp() + ":" + this.userPort;
        try {
            try {
                DataInputStream open =new DataInputStream( getAssets().open("web/package.json"));
                byte[] bArr = new byte[open.available()];
                open.readFully(bArr);
                JSONObject jSONObject = new JSONObject(new String(bArr));
                ServerPackage serverPackage = new ServerPackage();
                serverPackage.setLogo(jSONObject.getString("logo"));
                serverPackage.setName(jSONObject.getString("name"));
                serverPackage.setVersion(jSONObject.getString("version"));
                serverPackage.setAuthor(jSONObject.getString("author"));
                serverPackage.setDescription(jSONObject.getString("description"));
                serverPackage.setMain(jSONObject.getString("main"));
                serverPackage.setServerPath("web");
                open.close();
                ServerJS.getInstance().init(serverPackage, new JsEvaluator(this));
                serverPackage.setMiniJsServer(new MiniJsServer(this.userPort, serverPackage));
                try {
                    serverPackage.getMiniJsServer().start();
                } catch (IOException e2) {
                    e2.printStackTrace();
                }
                serverPackage.setPort(str);
                serverPackage.setPower(true);
            } catch (JSONException e3) {
                e3.printStackTrace();
            }
        } catch (IOException e4) {
            e4.printStackTrace();
        }
    }

    @Override // com.xlzhen.sharefileserver.service.BaseService, android.app.Service
    public IBinder onBind(Intent intent) {
        return new Binder();
    }

    @Override // com.xlzhen.sharefileserver.service.BaseService, android.app.Service
    public int onStartCommand(Intent intent, int i, int i2) {
        startManagementServer();
        startUserServer();
        notification();
        return super.onStartCommand(intent, Service.START_FLAG_REDELIVERY, i2);
    }

    public void notification() {
        NotificationChannel notificationChannel = Build.VERSION.SDK_INT >= 26 ? new NotificationChannel(ServerRunService.class.getName(), ServerRunService.class.getSimpleName(), NotificationManager.IMPORTANCE_HIGH) : null;
        NotificationManager notificationManager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
        if (Build.VERSION.SDK_INT >= 26) {
            notificationManager.createNotificationChannel(notificationChannel);
        }
        Notification.Builder builder = new Notification.Builder(getApplicationContext());
        if (Build.VERSION.SDK_INT >= 26) {
            builder.setChannelId(ServerRunService.class.getName());
        }
        builder.setContentText(Html.fromHtml(getString(R.string.notification_content)));
        builder.setContentTitle(getString(R.string.notification_title));
        builder.setSmallIcon(R.mipmap.ic_launcher_notification);
        if (Build.VERSION.SDK_INT >= 23) {
            builder.setLargeIcon(Icon.createWithResource(getApplicationContext(), (int) R.mipmap.ic_launcher));
        }
        builder.setContentIntent(PendingIntent.getActivity(getApplicationContext(), 1, getPackageManager().getLaunchIntentForPackage(getPackageName()), PendingIntent.FLAG_IMMUTABLE));
        Notification build = builder.build();
        build.flags = 2;
        build.flags = 34;
        build.flags = 34 | 64;
        startForeground(1, build);
    }
}

File: app/src/main/java/com/xlzhen/sharefileserver/service/ServerRunService.kt
package com.xlzhen.sharefileserver.service

import android.app.Notification
import android.app.NotificationChannel
import android.app.NotificationManager
import android.app.PendingIntent
import android.app.Service
import android.content.Context
import android.content.Intent
import android.content.pm.ServiceInfo
import android.graphics.drawable.Icon
import android.os.Binder
import android.os.Build
import android.os.IBinder
import android.text.Html
import com.evgenii.jsevaluator.JsEvaluator
import com.xlzhen.sharefileserver.R
import com.xlzhen.sharefileserver.entity.ServerPackage
import com.xlzhen.sharefileserver.server.MiniJsServer
import com.xlzhen.sharefileserver.server.ServerJS
import com.xlzhen.sharefileserver.utils.NetWorkUtils
import org.json.JSONException
import org.json.JSONObject
import java.io.DataInputStream
import java.io.IOException

class ServerRunService : BaseService() {
    var rootPort = 8080
    var userPort = 8090

    fun startManagementServer() {
        host = "http://${NetWorkUtils.getDeviceIp()}:$rootPort"
        try {
            val open = DataInputStream(assets.open("management/package.json"))
            val bArr = ByteArray(open.available())
            open.readFully(bArr)
            val jsonObject = JSONObject(String(bArr))
            val serverPackage = ServerPackage().apply {
                logo = jsonObject.getString("logo")
                name = jsonObject.getString("name")
                version = jsonObject.getString("version")
                author = jsonObject.getString("author")
                description = jsonObject.getString("description")
                main = jsonObject.getString("main")
                serverPath = "management"
            }
            open.close()

            ServerJS.instance!!.init(serverPackage, JsEvaluator(this))
            serverPackage.miniJsServer = MiniJsServer(rootPort, serverPackage)

            try {
                serverPackage.miniJsServer?.start()
            } catch (e2: IOException) {
                e2.printStackTrace()
            }
            serverPackage.port = host
            serverPackage.isPower = true
        } catch (e3: IOException) {
            e3.printStackTrace()
        } catch (e4: JSONException) {
            e4.printStackTrace()
        }
    }

    fun startUserServer() {
        val str = "http://${NetWorkUtils.getDeviceIp()}:$userPort"
        try {
            val open = DataInputStream(assets.open("web/package.json"))
            val bArr = ByteArray(open.available())
            open.readFully(bArr)
            val jsonObject = JSONObject(String(bArr))
            val serverPackage = ServerPackage().apply {
                logo = jsonObject.getString("logo")
                name = jsonObject.getString("name")
                version = jsonObject.getString("version")
                author = jsonObject.getString("author")
                description = jsonObject.getString("description")
                main = jsonObject.getString("main")
                serverPath = "web"
            }
            open.close()

            ServerJS.instance!!.init(serverPackage, JsEvaluator(this))
            serverPackage.miniJsServer = MiniJsServer(userPort, serverPackage)

            try {
                serverPackage.miniJsServer?.start()
            } catch (e2: IOException) {
                e2.printStackTrace()
            }
            serverPackage.port = str
            serverPackage.isPower = true
        } catch (e4: IOException) {
            e4.printStackTrace()
        } catch (e3: JSONException) {
            e3.printStackTrace()
        }
    }

    override fun onBind(intent: Intent?): IBinder? {
        return Binder()
    }

    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
        startManagementServer()
        startUserServer()
        notification()
        return super.onStartCommand(intent, Service.START_FLAG_REDELIVERY, startId)
    }

    fun notification() {
        val notificationManager = getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            val notificationChannel = NotificationChannel(
                ServerRunService::class.java.name,
                ServerRunService::class.java.simpleName,
                NotificationManager.IMPORTANCE_HIGH
            )
            notificationManager.createNotificationChannel(notificationChannel)
        }
        val builder = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            Notification.Builder(applicationContext, ServerRunService::class.java.name)
        } else {
            @Suppress("DEPRECATION")
            Notification.Builder(applicationContext)
        }

        val htmlText = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
            Html.fromHtml(getString(R.string.notification_content), Html.FROM_HTML_MODE_LEGACY)
        } else {
            @Suppress("DEPRECATION")
            Html.fromHtml(getString(R.string.notification_content))
        }
        builder.setContentText(htmlText)
        builder.setContentTitle(getString(R.string.notification_title))
        builder.setSmallIcon(R.mipmap.ic_launcher_notification)

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            builder.setLargeIcon(Icon.createWithResource(applicationContext, R.mipmap.ic_launcher))
        }

        val intent = packageManager.getLaunchIntentForPackage(packageName)
        val pendingIntent = PendingIntent.getActivity(applicationContext, 1, intent, PendingIntent.FLAG_IMMUTABLE)
        builder.setContentIntent(pendingIntent)

        val build = builder.build()
        build.flags = build.flags or Notification.FLAG_ONGOING_EVENT or Notification.FLAG_FOREGROUND_SERVICE

        if (Build.VERSION.SDK_INT >= 34) {
            startForeground(1, build, ServiceInfo.FOREGROUND_SERVICE_TYPE_DATA_SYNC)
        } else {
            startForeground(1, build)
        }
    }

    companion object {
        var f284c: ServerRunService? = null
        var host: String? = null
            private set

        @JvmStatic
        fun start(context: Context) {
            if (f284c == null) {
                f284c = ServerRunService()
                context.startService(Intent(context, ServerRunService::class.java))
            }
        }
    }
}
